duckduckgo/iOS
# [DuckPlayer] - 2. Overlay Views

Task/Issue URL: https://app.asana.com/0/1204099484721401/1207650110399906/f

**Description**:
- Implements Overlay Script for iOS
- Allows user to see video inline or open Duck Player
- Cleans up UserScripts (messaging) to match with what we do in other places on iOS (like subscriptions)

ℹ️   Known issue:  The FE is sometimes starting navigation instead of pushing a WKMessage, which causes you to have double tap the "Watch in DuckPlayer" button sometimes.   This is [reported](https://app.asana.com/0/1204099484721401/1207658032936628/f) and should be fixed as part of the FE work



**Steps to test this PR**:

⚠️  0. Update privacy config in Debug to use: https://jsonblob.com/1253721903368888320 and restart the app

1.  Go to Settings > Duck Player, and Select "Ask every time"
2.  Open a New tab and go to Youtube.com
3. Tap on a video
4. Confirm the Overlay shows on top of the video
5. Tap Watch here, and confirm video starts playing
6. Refresh the page
7. Tap Watch in Duck Player and confirm video plays in DP
8. Hit the back button
9. Confirm you're taken to the video page (not the youtube page)
